warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownNegationVisitor.java

```diff
@@ -231,19 +234,47 @@ public static JexlNode applyDeMorgans(JexlNode root, boolean negateRoot) {
         return newRoot;
     }
     
+    /**
+     * Negations take several forms in the query tree. Thus, some work needs to be done in order to drop all constituent parts of the negation.
+     *
+     * @param toDrop
+     *            an ASTNotNode
+     * @param state
+     *            the current negation state
+     * @return the replacement node
+     */
     private JexlNode dropNot(ASTNotNode toDrop, NegationState state) {
-        JexlNode newNode = toDrop.jjtGetChild(0);
+        JexlNode child = toDrop.jjtGetChild(0);
         
         // no need to swap nodes if toDrop is the root
         if (toDrop.jjtGetParent() != null) {
+            
+            // check for adjacent not nodes
+            if (child instanceof ASTNotNode || QueryPropertyMarker.findInstance(child).isAnyType()) {
```

Comment:
What if the child of the child is a Not Node?  What if the child of the child of the child is a NotNode?  Doesn't the visitor pattern do the appropriate thing as it moves down the tree?

Reply:
This is handled by the calling method. After dropping the negation it immediately visits the returned node. I believe several of the tests for this visitor cover the exact case you're thinking of.